### ISSUE TITLE
Content Hub - Incomplete addresses are sent resulting in Null Farms

### ISSUE THREAD

**Issue by joewhitsitt** [uiowa/uiowa#5719]
# Problem
Within content_hub_core/inc/content_hub_core.event_sync.inc for node insert/update there is code that is sending the value of the address field off to events.uiowa.edu but is not checking to see if all values within the compound field are populated.

```php
$address = $wrapper->field_address->value();
$event['location']['address'] = $address['thoroughfare'] . ', ' . $address['locality'] . ', ' . $address['administrative_area'] . ' ' . $address['postal_code'];
```
This ends up being passed as `,,` and geolocated to Null Farms.. 🤣 

![event calendar](https://user-images.githubusercontent.com/4663676/191618014-b27817d3-973a-405b-8b64-749bbcab73c9.JPG)

The CSI group has to go in and correct these on a case-by-case basis. They could add some regex to account for this, but I think it would be better if we addressed it on our side.

# Proposed solution
- [ ] If address sub fields are optional, account for them with proper checks in code.

**Comment by bspeare** [comment]
Are we able to test this fix out on their test server?  If so, does anyone recall what that url is?

**Comment by joewhitsitt** [comment]
I want to say it is https://events-test.uiowa.edu/. Could also coordinate with Jeremy to say it is ready to test

**Comment by bspeare** [comment]
I emailed Jeremy on Tuesday and I still have not heard back from him.  I'm going to remove myself from this issue and move it back to planned.

**Comment by briand44** [comment]
I'm not sure what we are waiting on Jeremy for? I was able to create an event on content.dev.drupal.uiowa.edu and have it appear on events-test.uiowa.edu.

https://events-test.uiowa.edu/33102

**Comment by briand44** [comment]
Seems like we should be able to check out the feature branch on 702 dev and move forward with testing this.

**Comment by joewhitsitt** [comment]
DEV is on tags/2022-09-26 instead of master or feature_null_farms_fix.

**Comment by briand44** [comment]
>DEV is on tags/2022-09-26 instead of master or feature_null_farms_fix.

Right, that is why I mentioned

>we should be able to check out the feature branch on 702 dev

I am not going to have time to test this now. Have a meeting shortly.

**Comment by briand44** [comment]
Oh I get your point. It should not be on a tag, it should be on master if we are not checking out feature branch. Good catch!

**Comment by bspeare** [comment]
Yes, I switched it back to the wrong branch this morning.  Thanks for catching that!

**Comment by bspeare** [comment]
> Seems like we should be able to check out the feature branch on 702 dev and move forward with testing this.

You are probably right.  I think I would this would be a issue to pair on to get a better sense of how to test it on https://events-test.uiowa.edu/.

**Comment by joewhitsitt** [comment]
Planning a ~11ish mob on this today

**Comment by joewhitsitt** [comment]
Fixed by commit 3bdc1f346 where we are checking for thoroughfare before constructing the address to pass to events.uiowa.edu for both address and place. This has been deployed to PROD
